wp-directives.php

```diff
@@ -36,3 +37,23 @@ function wp_directives_register_scripts()
 }
 
 add_action('init', 'wp_directives_register_scripts');
+
+function add_wp_link_attribute($block_content)
+{
+	$site_url = parse_url(get_site_url());
+	$w = new WP_HTML_Tag_Processor($block_content);
+	while ($w->next_tag('a')) {
+		$link = parse_url($w->get_attribute('href'));
+
+		if ($w->get_attribute('target') === '_blank') {
+			break;
+		}
+
+		if (is_null($link['host']) || ($link['host'] ===  $site_url['host'])) {
+			$w->set_attribute('wp-link', 'true');
+		}
+	};
+	return $w;
```

Comment:
I'm not 100% sure how is the final API, but I think it's safer to do this. If not, you'd be returning the `WP_HTML_Tag_Processor` instance instead:

```suggestion
	return (string) $w;
```

_The `(string) $variable` thing is similar to JavaScript's `.toString()`._

Reply:
Changed 🙂